packages/hurl/src/runner/entry.rs

```diff
@@ -117,6 +117,8 @@ pub fn run(
     // 3. finally, run the remaining asserts
     let mut asserts = vec![];
 
+    let max_filesize = client_options.max_filesize;
```

Comment:
Instead of passing the max_filesize up to the result, we need to pass it to the run execution. Libcurl has a function to support it (https://curl.se/libcurl/c/CURLOPT_MAXFILESIZE.html) that can be used in the client (see the methodes on hanlde in HTTP client). This way we dont need to check the file size when we save it: libcurl will stop the download and the run will be in error.

It's simpler for us (check is delegated to curl). And for instance, if the user limits the max_filesize to 2 bytes, and the file is 20G, the HTTP engine don't transfer 20 GB and stop as soon as it get 2 bytes.